compiler/rustc_mir_build/src/build/matches/util.rs

```diff
@@ -24,6 +24,51 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             .collect()
     }
 
+    #[instrument(skip(self), level = "debug")]
+    pub(crate) fn field_match_pairs_tuple_struct<'pat>(
+        &mut self,
+        place_builder: PlaceBuilder<'tcx>,
+        subpatterns: &'pat [FieldPat<'tcx>],
+    ) -> Vec<MatchPair<'pat, 'tcx>> {
+        let place_ty = place_builder
+            .try_ty(&self.local_decls, self)
+            .map(|ty| self.tcx.normalize_erasing_regions(self.param_env, ty));
+        debug!(?place_ty);
+
+        subpatterns
+            .iter()
+            .map(|fieldpat| {
+                // NOTE: With type ascriptions it can happen that we get errors
+                // during borrow-checking on higher-ranked types if we use the
+                // ascribed type as the field type, so we try to get the actual field
+                // type from the `Place`, if possible, see issue #96514
```

Comment:
if my understanding is correct, this "if possible" is quite right, as getting the correct field type isn't relevant if `try_ty` returns `None`.